public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.test.tsx

```diff
@@ -0,0 +1,147 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { FormApi } from 'final-form';
+import React from 'react';
+import { Form } from 'react-final-form';
+import { Provider } from 'react-redux';
+import selectEvent from 'react-select-event';
+
+import { InventoryService } from 'app/percona/inventory/Inventory.service';
+import { nodesMockMultipleAgentsNoPMMServer, nodesMock } from 'app/percona/inventory/__mocks__/Inventory.service';
+import * as NodesReducer from 'app/percona/shared/core/reducers/nodes/nodes';
+import { configureStore } from 'app/store/configureStore';
+
+import { NodesAgents } from './NodesAgents';
+
+const fetchNodesActionActionSpy = jest.spyOn(NodesReducer, 'fetchNodesAction');
+
+describe('Nodes Agents:: ', () => {
+  let formAPI: FormApi<any>;
+  const submitMock = jest.fn();
+
+  function setup() {
+    render(
+      <Provider store={configureStore()}>
+        <Form
+          onSubmit={submitMock}
+          render={({ handleSubmit, form }) => {
+            formAPI = form;
+            return (
+              <form data-testid="node-agents-form" onSubmit={handleSubmit}>
+                <NodesAgents form={form} />
+              </form>
+            );
+          }}
+        />
+      </Provider>
+    );
+  }
+
+  it('should not pick any agent when the selected node is not pmm-server', async () => {
+    const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+
+    jest
+      .spyOn(InventoryService, 'getNodes')
+      .mockReturnValue(Promise.resolve({ nodes: nodesMockMultipleAgentsNoPMMServer }));
+    setup();
+    await waitFor(() => {
+      expect(fetchNodesActionActionSpy).toHaveBeenCalled();
+    });
+
+    const nodesSelect = screen.getByLabelText('Nodes');
+    await selectEvent.select(nodesSelect, [nodesMockMultipleAgentsNoPMMServer[0].node_name], {
```

Comment:
The console error comes from not await the select event:

```tsx
await waitFor(() => selectEvent.select(nodesSelect [nodesMockMultipleAgentsNoPMMServer[0].node_name], {
  container: document.body,
}));
```

After it has been wrapped in `waitFor` no console errors are shown.